### DISCUSSION TITLE
using Dash Design Kit (ddk.graph)

### DISCUSSION THREAD

**Discussion by malammar0** [predict-idlab/plotly-resampler#130] [Q&A] [answered]
Hi!
We've been using dcc.Graph with our plotly-resampler code and everything has been working smoothly. But we're trying to use a Dash Design Kit graph (ddk.graph), we get this error:
**TraceUpdater: no graphs with ID="graph" found**
Here is our code, the only thing we changed was dcc->ddk
ddk.Graph(
        id="graph",
        figure=fig,
        style={},
        config={
            "displayModeBar": True,
            "displaylogo": False,
        },
    ),
TraceUpdater(id="trace-updater", gdID="graph")

Would appreciate it if you had any tips/comments!

**Comment by malammar0** [comment]
Hi, upon further investigation this seems to be the issue: 
let graphDiv = document?.querySelectorAll('div[id*="' + gdID + '"][class="dash-graph"]');
https://github.com/predict-idlab/trace-updater/blob/944e9272ab1931a99fd49f3436a0e5293f1bb29b/src/lib/components/TraceUpdater.react.js#L84

Changing [class="dash-graph"]" to [class="dash-graph ddk-graph "] in the trace-updater.min.js makes it work. (please note the space after ddk-graph)
![image](https://user-images.githubusercontent.com/108471576/198361220-204b7f48-fa11-4c78-b85f-03009ab3a247.png)

**Comment by jonasvdd** [comment]
Hi @malammar0, thank yo for submitting this issue! 

It indeed seems that your suggestion should resolve this! I will create an issue for it, and mention when this is resolved within traceupdater.

**Comment by jonasvdd** [comment, marked as answer]
Hi @malammar0, this should be fixed in `trace-updater==0.0.9`, can you verify this? (you will need to manually install this version of trace-updater)

```bash
pip install trace-updater==0.0.9
```

**Reply by jonasvdd** [reply]
Hi @malammar0, have you found any time to look into this?

**Reply by jonasvdd** [reply]
Hi @malammar0, any updates whether the latest traceupdater version resolves this?

**Reply by malammar0** [reply]
Hi Jonas sorry for the late reply, yes the new update worked perfectly! 
Thanks!
